Cosmetics.py

```diff
@@ -969,6 +969,27 @@ def patch_dpad_left(rom: Rom, settings: Settings, log: CosmeticsLog, symbols: di
         rom.write_byte(symbols['CFG_DPAD_ON_THE_LEFT'], 0x00)
     log.display_dpad = settings.display_dpad
 
+def patch_song_names(rom: Rom, settings: Settings, log: CosmeticsLog, symbols: dict[str, int]) -> None:
+    bytes_to_write = []
+    if settings.display_custom_song_names != 'off':
+        if settings.display_custom_song_names == 'top':
+            rom.write_byte(symbols['CFG_SONG_NAME_POSITION'], 0x00)
+        if settings.display_custom_song_names == 'pause':
+            rom.write_byte(symbols['CFG_SONG_NAME_POSITION'], 0x01)
+
+
+        for area, song_name in log.bgm.items():
+            text_bytes = []
+            if (len(song_name) > 50):
+                song_name_cropped = song_name[:50]
+                text_bytes = list(ord(c) for c in song_name_cropped)
+            else:
+                text_bytes[0:len(song_name)] = list(ord(c) for c in song_name)
+                text_bytes[len(song_name):50] = [ord('\0')] * (50 - len(song_name))
+            bytes_to_write += text_bytes
```

Comment:
I think it would be nice to run this code even if the cosmetic setting is off, to allow emulator plugins to read the song name, e.g. for displaying on a stream layout or creating a chat command to request the currently playing song.

Reply:
No objection for me, maybe put this part in a separate function that can be called at will?

Reply:
I'm not sure what you're suggesting. Wouldn't unindenting the `for` loop accomplish this?

Reply:
Oh yeah i completely misinterpreted, my bad.
So fill the SONG_NAMES area in something like Patches.py in all situations, and move it out of cosmetic context in config.asm?

Reply:
I think keeping it in the cosmetic context would be good, since it's discoverable and versioned, making it easier for plugins to work with.

Reply:
I've included my suggestion in code form in my new review.